src/index.ts

```diff
@@ -131,48 +146,77 @@ export default class SwupPreloadPlugin extends Plugin {
 		const el = event.delegateTarget;
 		if (!(el instanceof HTMLAnchorElement)) return;
 
-		this.preloadLink(el);
+		this.preload(el, { priority: true });
 	};
 
-	preloadLink(el: HTMLAnchorElement) {
-		const { url, href } = Location.fromElement(el);
+	async preload(url: string, options?: PreloadOptions): Promise<PageData | void>;
+	async preload(urls: string[], options?: PreloadOptions): Promise<PageData[]>;
+	async preload(el: HTMLAnchorElement, options?: PreloadOptions): Promise<PageData | void>;
+	async preload(
+		link: string | string[] | HTMLAnchorElement,
+		options: PreloadOptions = {}
+	): Promise<PageData | (PageData | void)[] | void> {
+		let url: string;
+		let trigger: HTMLAnchorElement | undefined;
+		const priority = options.priority ?? false;
+
+		// Allow passing in array of elements or urls
+		if (Array.isArray(link)) {
+			return Promise.all(link.map((url) => this.preload(url)));
+		}
+		// Allow passing in an anchor element
+		else if (link instanceof HTMLAnchorElement) {
+			trigger = link;
+			({ url } = Location.fromElement(link));
+		}
+		// Allow passing in a url
+		else {
+			url = String(link);
+		}
 
-		// Bail early if the visit should be ignored by swup
-		if (this.swup.shouldIgnoreVisit(href, { el })) return;
+		if (!this.shouldPreload(url, trigger)) {
+			return;
+		}
 
-		// Bail early if the link points to the current page
-		if (url === getCurrentUrl()) return;
+		return new Promise<PageData | void>((resolve) => {
+			this.queue.add(() => {
+				const preloadPromise = this.performPreload(url);
+				this.preloadPromises.set(url, preloadPromise);
+				preloadPromise
+					.catch(() => {})
+					.then((page) => resolve(page))
+					.finally(() => {
+						this.queue.next();
+						this.preloadPromises.delete(url);
+					});
+			}, priority);
+		});
+	}
 
-		// Bail early if the page is already in the cache
-		if (this.swup.cache.has(url)) return;
+	preloadLinks() {
+		const selector = 'a[data-swup-preload], [data-swup-preload-all] a';
+		const links = Array.from(document.querySelectorAll<HTMLAnchorElement>(selector));
+		links.forEach((el) => this.preload(el));
+	}
 
-		// Bail early if there is already a preload running
-		if (this.preloadPromises.has(url)) return;
+	protected shouldPreload(location: string, el?: HTMLAnchorElement): boolean {
+		const { url, href } = Location.fromUrl(location);
 
-		// Bail early if there are more then the maximum concurrent preloads running
-		if (this.preloadPromises.size >= this.options.throttle) return;
+		// Already in cache?
+		if (this.swup.cache.has(url)) return false;
+		// Already preloading?
+		if (this.preloadPromises.has(url)) return false;
+		// Should be ignored anyway?
+		if (this.swup.shouldIgnoreVisit(href, { el })) return false;
+		// Special condition for links: points to current page?
+		if (el && url === getCurrentUrl()) return false;
```

Comment:
This should check for the same resolved URL. We should probably make `swup.isSameResolvedUrl()` public and use it here